### PR TITLE
Get/api/topics

### PR DESCRIPTION
**Pull request by DevDolphin7**
Install supertest (dev) and express, complete GET /api/topics endpoint, error handle "not-a-route"

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+describe("/api/topics", () => {
+  describe("GET", () => {
+    test("Returns an object with topics key", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.topics).toBeTruthy();
+        });
+    });
```
Comment (nc-rayhaan) [inline __tests__/app.test.js:31]: This test isn't really needed as the test below proves that the body is truthy

File: endpoints/app.endpoints.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const { getTopics } = require("../controllers/topics.controller");
+
+const app = express();
+
+app.get("/api/topics", getTopics);
+
+app.all("*", (request, response, next) => {
+  const error = { status: 404, message: "Not a valid route" };
+  customError(error, request, response, next);
+});
```
Comment (nc-rayhaan) [inline endpoints/app.endpoints.js:12]: So this one isn't actually a custom error as you are not forcing the error to happen. Therefore, you can send your response directly in the callback without invoking `customError`. You can then use `customError` for when that is needed in a future ticket 😄

File: models/topics.model.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const db = require("../db/connection");
+
+exports.fetchTopics = () => {
+  return db.query("SELECT * FROM topics").then(({ rows }) => {
+    console.log(rows)
```
Comment (nc-rayhaan) [inline models/topics.model.js:5]: Please remove console logs before pushing!

**Review by nc-rayhaan — COMMENTED** [review]
Nice start! Left a couple comments for you for when you get a chance

File: models/topics.model.js (hunk 3 above)
Comment (DevDolphin7) [reply models/topics.model.js:5]: Sorry!

**Pull request merged**